sklearn/utils/tests/test_stats.py

```diff
@@ -1,43 +1,61 @@
 import numpy as np
+import pytest
 from numpy.testing import assert_allclose
 from pytest import approx
 
+from sklearn.utils._array_api import (
+    yield_namespace_device_dtype_combinations,
+    yield_namespaces,
+)
+from sklearn.utils.estimator_checks import _array_api_for_tests
 from sklearn.utils.stats import _weighted_percentile
 
 
-def test_weighted_percentile():
-    y = np.empty(102, dtype=np.float64)
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+)
+def test_weighted_percentile(array_namespace, device, dtype_name):
+    xp = _array_api_for_tests(array_namespace, device)
+
+    y = xp.empty(102, dtype=xp.float64, device=device)
     y[:50] = 0
     y[-51:] = 2
     y[-1] = 100000
     y[50] = 1
-    sw = np.ones(102, dtype=np.float64)
+    sw = xp.ones(102, dtype=xp.float64, device=device)
     sw[-1] = 0.0
     score = _weighted_percentile(y, sw, 50)
```

Comment:
We need to turn on array API support before calling `_weighted_percentile` (also in the other tests) to see the effect of it. Without enabling it `get_namespace` will always return a (wrapped) Numpy namespace.

```suggestion
    with config_context(array_api_dispatch=True):
        score = _weighted_percentile(y, sw, 50)
```

In other tests we also call the thing we want to test with array API support turned off and then later compare the results. Given we have `assert approx(score) == 1` below we might not need to do that. However we probably do want a test with array API support turned off and "vanilla numpy arrays"  as input. To make sure that the changes we've made don't break things.